src/parallel/boundaries/BoundaryUtils.h

```diff
@@ -0,0 +1,136 @@
+/*
+ * BoundaryUtils.h
+ *
+ *  Created on: 24 March 2023
+ *      Author: amartyads
+ */
+
+#pragma once
+
+#include <array>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include "molecules/Molecule.h"
+
+/**
+ * Includes enums and helper functions for processing boundary conditions.
+ *
+ * Does not actually work on any molecule containers, hence can be used
+ * in a general sense for other calculations.
+ */
+namespace BoundaryUtils {
+
+/**
+ * enum storing the types of boundaries currently supported.
+ *
+ * The currently supported boundary types are
+ * PERIODIC - default behaviour, periodic boundaries
+ * OUTFLOW - molecules exiting the boundary are deleted
+ * REFLECTING - molecules exiting the boundary have their velocities
+ * reversed in the direction they are leaving
+ * ERROR - kept for sanity checks
+ *
+ * This can be extended if needed.
+ */
+enum class BoundaryType { PERIODIC, OUTFLOW, REFLECTING, ERROR };
```

Comment:
I'm reverting this, because while rewriting my documentation I realized that `BoundaryUtils` was always supposed to have global boundaries and never local. Local boundaries are never stored. I first check whether a wall is a global wall, and then check the type of boundary.